src/background/index.js

```diff
@@ -1,28 +1,60 @@
-import { TEXT_SCAN } from '../extensionMessageTypes';
+import {
+    CONTEXT_MENU_TITLE,
+    LOOKUP_PLACE_TITLE,
+    SCAN_PARAGRAPH_TITLE,
+    CONTEXT_MENU_ID,
+    LOOKUP_PLACE_ID,
+    SCAN_PARAGRAPH_ID,
+} from './constants.js';
+import { LOOKUP_PLACE, SCAN_PARAGRAPH } from '../extensionMessageTypes';
 
-function selectionTextListener(info) {
-    chrome.tabs.query(
-        {
-            active: true,
-            currentWindow: true,
-        },
-        (tabs) => {
-            chrome.tabs.sendMessage(tabs[0].id, {
-                message: TEXT_SCAN,
-                data: info.selectionText,
-            }); // Handle The Response
-        },
-    );
+function lookupPlaceListener({ selectionText }) {
+    chrome.tabs.query({
+        active: true,
+        currentWindow: true,
+    },
+    ([firstTab]) => {
+        chrome.tabs.sendMessage(firstTab.id, {
+            message: LOOKUP_PLACE,
+            data: selectionText,
+        }); // Handle The Response
+    });
+}
+
+function scanParagraphListener({ selectionText }) {
+    chrome.tabs.query({
+        active: true,
+        currentWindow: true,
+    },
+    ([firstTab]) => {
+        chrome.tabs.sendMessage(firstTab.id, {
+            message: SCAN_PARAGRAPH,
+            data: selectionText,
+        }); // Handle The Response
+    });
 }
 
 const listenerFunctionMap = {
-    selectionWanderland: selectionTextListener,
+    [LOOKUP_PLACE_ID]: lookupPlaceListener,
+    [SCAN_PARAGRAPH_ID]: scanParagraphListener,
 };
 
 chrome.runtime.onInstalled.addListener(() => {
     chrome.contextMenus.create({
-        id: 'selectionWanderland',
-        title: 'Lookup Place',
+        id: CONTEXT_MENU_ID,
+        title: CONTEXT_MENU_TITLE,
+        contexts: ['selection'],
+    });
+    chrome.contextMenus.create({
+        id: LOOKUP_PLACE_ID,
+        parentId: CONTEXT_MENU_ID,
+        title: LOOKUP_PLACE_TITLE,
+        contexts: ['selection'],
+    });
+    chrome.contextMenus.create({
+        id: SCAN_PARAGRAPH_ID,
+        parentId: CONTEXT_MENU_ID,
+        title: SCAN_PARAGRAPH_TITLE,
         contexts: ['selection'],
     });
 });
```

Comment:
This comment is about the last lines in the file... I like what you did with the map but it makes you have to make two functions that do the same thing. What if you made the two functions into one: ```selectionTextListener``` and removed the map and made the last lines...
```
chrome.contextMenus.onClicked.addListener((info, tab) => {
   const message = info.menuItemId === SCAN_PARAGRAPH_ID ? SCAN_PARAGRAPH : LOOKUP_PLACE;
   selectionTextListener(info, message)
});
```

Reply:
If we add any more functions into the context menus, then we'd have to change this. I could come up with some other way to reduce the bloat though.

Reply:
yeah you're right. it was just an idea. the mind point is the two functions identical with just one constant different

Reply:
let me know if the new way is better. I made it so that it's easy enough to change if we ever need separate functions for each